Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCGLogger
+import Deferred
+
+/// Small enum to help strength parameter requirements for potential bookmark highlights query below.
+private enum BookmarkSource {
+    case Mirror
+    case Local
+
+    var modifiedColumn: String  {
+        switch self {
+        case Mirror: return "server_modified"
+        case Local: return "local_modified"
+        }
+    }
+
+    var tableName: String {
+        switch self {
+        case Mirror: return TableBookmarksMirror
+        case Local: return TableBookmarksLocal
+        }
+    }
+}
+
+extension SQLiteHistory: HistoryRecommendations {
+    public func getHighlights() -> Deferred<Maybe<Cursor<Site>>> {
+        let limit = 19
+        let microsecondsPerMinute: Timestamp = 60_000_000 // 1000 * 1000 * 60
+        let thirtyMinutesAgo = NSDate.nowMicroseconds() - 30 * microsecondsPerMinute
+        let threeDaysAgo = NSDate.nowMicroseconds() - (60 * microsecondsPerMinute) * 24 * 3
+        let bookmarkLimit = 1
+        let historyLimit = limit - bookmarkLimit
+
+        let recentHistory = [
+            "SELECT *",
+            "FROM (",
+            "   SELECT \(TableHistory).id AS historyID, url, title, guid, sum(1) AS visitCount, max(\(TableVisits).date) AS visitDate",
+            "   FROM \(TableHistory)",
+            "   LEFT JOIN \(TableVisits) ON \(TableVisits).siteID = \(TableHistory).id",
+            "   WHERE title NOT NULL AND title != '' AND is_deleted = 0",
+            "   GROUP BY url",
+            "   ORDER BY visitDate DESC",
+            "   LIMIT \(historyLimit)",
+            ")",
+            "WHERE visitCount <= 3 AND visitDate < \(thirtyMinutesAgo)",
+        ].joinWithSeparator(" ")
+
+        let bookmarkHighlights = [
+            "SELECT historyID, url, title, guid, visitCount, visitDate",
+            "FROM (",
+            "   SELECT *",
```

Comment:
You could do something like this:

```
    private let localBookmarksView =
    "CREATE VIEW \(ViewBookmarksLocalOnMirror) AS " +
    "SELECT -1 AS id, guid, type, is_deleted, parentid, parentName, feedUri, siteUri, pos, title, description, bmkUri, folderName, faviconID, NULL AS local_modified, server_modified, 0 AS is_overridden " +
    "FROM \(TableBookmarksMirror) WHERE is_overridden IS NOT 1 " +
    "UNION ALL " +
    "SELECT -1 AS id, guid, type, is_deleted, parentid, parentName, feedUri, siteUri, pos, title, description, bmkUri, folderName, faviconID, local_modified, NULL AS server_modified, 1 AS is_overridden " +
    "FROM \(TableBookmarksLocal) WHERE is_deleted IS NOT 1"
```

or combine the two into a single `modified` column and and check `is_overridden` at point of use if you want to know whether it's a local or server time, and don't ever want both the local modification and server modification time for a locally modified bookmark.